src/MetalFunFactoryV2.sol

```diff
@@ -172,7 +172,7 @@ contract MetalFunFactoryV2 is Ownable, ERC721Holder {
         uint256 _totalSupply,
         address _recipient,
         uint256 _recipientAmount
-    ) public returns (InstantLiquidityToken token, uint256 lpTokenId) {
+    ) public returns (InstantLiquidityToken token, uint256 lpTokenId, address tokenAddress) {
```

Comment:
why are you doing returning this value again? it is already returned as InstantLiquidityToken.